Apps/APN/App.js

```diff
@@ -1,263 +1,102 @@
-    bgTasksValue,
-    initialiseCioPackage,
-  ]);
-
-  const fetchConfigsOrSetDefault = async () => {
-    const keyStorageObj = new CioKeyValueStorage();
-    const bgDelayValue = await keyStorageObj.getBGQSecondsDelay();
-    const bgTasksValue = await keyStorageObj.getBGQMinTasksInQueue();
-    const screenTrackValue = await keyStorageObj.getScreenTrack();
-    const deviceAttrValue = await keyStorageObj.getDeviceAttributesTrack();
-    const debugModeValue = await keyStorageObj.getDebugModeConfig();
-    const trackUrl = await keyStorageObj.getTrackingUrl();
-
-    // Setting values here to show default values on Settings screens
-    if (screenTrackValue === null) {
-      await keyStorageObj.saveScreenTrack(DefaultConstants.SCREEN_TRACK_STATUS);
-    }
-    if (bgDelayValue === null) {
-      await keyStorageObj.saveBGQSecondsDelay(
-        `${DefaultConstants.BGQ_SECONDS_DELAY}`
-      );
-    }
-    if (bgTasksValue === null) {
-      await keyStorageObj.saveBGQMinTasksInQueue(
-        `${DefaultConstants.BGQ_MIN_TASKS_IN_QUEUE}`
-      );
-    }
-    if (deviceAttrValue === null) {
-      await keyStorageObj.saveDeviceAttributesTrack(
-        `${DefaultConstants.TRACK_DEVICE_ATTRIBUTES_STATUS}`
-      );
-    }
-    if (debugModeValue === null) {
-      await keyStorageObj.saveDebugModeConfig(
-        `${DefaultConstants.DEBUG_MODE_STATUS}`
-      );
-    }
-    setIsDeviceAttrTrackEnabled(
-      deviceAttrValue === null ? true : JSON.parse(deviceAttrValue)
-    );
-    setIsScreenTrackEnabled(
-      screenTrackValue === null ? true : JSON.parse(screenTrackValue)
-    );
-    setIsDebugModeEnabled(
-      debugModeValue === null ? true : JSON.parse(debugModeValue)
-    );
-    setBgDelayValue(
-      bgDelayValue === null
-        ? DefaultConstants.BGQ_SECONDS_DELAY
-        : parseInt(bgDelayValue)
-    );
-    setBgTasksValue(
-      bgTasksValue === null
-        ? DefaultConstants.BGQ_MIN_TASKS_IN_QUEUE
-        : parseInt(bgTasksValue)
-    );
-    setTrackingUrl(trackUrl);
-  };
+    const prepare = async () => {
+      const storageService = new StorageService();
+      const storage = await storageService.loadAll();
 
-  const initialiseCioPackage = useCallback(() => {
-    const configuration = new CustomerioConfig();
-    configuration.logLevel =
-      isDebugModeEnabled === null ? CioLogLevel.debug : isDebugModeEnabled;
-    configuration.autoTrackDeviceAttributes =
-      isDeviceAttrTrackEnabled === null ? true : isDeviceAttrTrackEnabled;
-    configuration.backgroundQueueMinNumberOfTasks = bgTasksValue;
-    configuration.backgroundQueueSecondsDelay = bgDelayValue;
-    if (trackingUrl != null) {
-      configuration.trackingApiUrl = trackingUrl;
-    }
-
-    const env = new CustomerIOEnv();
-    env.siteId = Env.siteId;
-    env.apiKey = Env.apiKey;
+      updateUserState(storage.user);
+      const sdkConfig = applyCustomerIoConfig(storage.sdkConfig);
+      updateCustomerIoSdkState(sdkConfig);
```

Comment:
This comment is similar to the comment I made about `CustomerIOService` class. 

I prefer the RN SDK init code before this PR because I find it easier to understand. 

I understand the code here for initializing the SDK, but I find it too advanced for someone browsing our sample app source code to learn how to use our SDK. 

I would prefer if the code inside of `App.js` uses SDK initialize code that looks identical to what our SDK docs show.

Reply:
Sorry if I didn't get your point here. `updateCustomerIoSdkState` is for updating state for UI component. I think you are suggesting to initialize SDK directly in `applyCustomerIoConfig`? If yes, I think we can discuss and resolve this in [above discussion](https://github.com/customerio/customerio-reactnative/pull/166#discussion_r1264070039).